### PR TITLE
Update notification

### PR DESCRIPTION
**Pull request by ghsteff**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
This updates the default `NotificationTheme`, adds an `info` variant, adds icons, and adds an optional action to Notification 

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

This updates `NotificationTheme`. Any app that overwrites or extends this component theme may have to update to match the new shape

## Other information

https://github.com/reaviz/reablocks/assets/40581813/70e8352d-7e68-4458-a64f-63ae965a1e81

Light mode variants
image

### CODE REVIEW

File: src/layers/Notification/Notifications.tsx [hunk 1]
```diff
@@ -94,19 +100,33 @@ export const Notifications: FC<NotificationsProps> = ({
...
+      notify(title, { variant: 'warning', icon: <WarningIcon />, ...options }),
     [notify]
   );
 
   const notifySuccess = useCallback(
     (title: string, options: NotificationOptions = {}) =>
-      notify(title, { ...options, variant: 'success' }),
+      notify(title, {
+        variant: 'success',
+        icon: <CheckCircleIcon />,
```
Comment (amcdnl) [inline src/layers/Notification/Notifications.tsx:121]: These need to be where you can pass them in

**Review by amcdnl — CHANGES_REQUESTED** [review]

File: src/layers/Notification/Notifications.tsx (hunk 1 above)
Comment (ghsteff) [reply src/layers/Notification/Notifications.tsx:121]: you can pass in whatever icon you want using the options in `notify`, `notifySuccess`, etc

```
notify("Notification Title", { icon: <CustomIcon />})
```

it'll overwrite this default icon through the spread `...options` below. Is that what you mean?

Comment (ghsteff) [reply src/layers/Notification/Notifications.tsx:121]: Pass the icons as optional props too?

```
<Notifications
  icons: {
    info: <InfoIcon />
    error: <ErrorIcon />
    warning: <WarningIcon />
    success: <SuccessIcon />
  }
>
```

**Comment by steppy452** [comment]
kind of a nit but can the light mode close button color be a little more contrasty?

**Comment by ghsteff** [comment]
> kind of a nit but can the light mode close button color be a little more contrasty?

Yea looks much better now thanks

**Pull request merged**